Src/FluentAssertions/AssertionExtensions.cs

```diff
@@ -230,13 +231,60 @@ public static StringCollectionAssertions Should(this IEnumerable<string> @this)
         }
 
         /// <summary>
-        /// Returns an <see cref="GenericDictionaryAssertions{TKey, TValue}"/> object that can be used to assert the
+        /// Returns an <see cref="GenericReadOnlyDictionaryAssertions{TKey, TValue}"/> object that can be used to assert the
         /// current <see cref="IDictionary{TKey, TValue}"/>.
         /// </summary>
         [Pure]
-        public static GenericDictionaryAssertions<TKey, TValue> Should<TKey, TValue>(this IDictionary<TKey, TValue> actualValue)
+        public static GenericReadOnlyDictionaryAssertions<TKey, TValue> Should<TKey, TValue>(this IDictionary<TKey, TValue> actualValue)
         {
-            return new GenericDictionaryAssertions<TKey, TValue>(actualValue);
+            IReadOnlyDictionary<TKey, TValue> dictionary = null;
+            if (actualValue != null)
+            {
+                dictionary = actualValue as IReadOnlyDictionary<TKey, TValue>;
+                if (dictionary == null)
+                    dictionary = new ReadOnlyDictionary<TKey, TValue>(actualValue);
```

Comment:
Again without testing, I think creating a new object breaks code like checking for reference equality.
```csharp
dictionary.Should().BeSameAs(dictionary)
```